server_fn_macro/src/lib.rs

```diff
@@ -744,6 +745,31 @@ impl ServerFnCall {
         }
     }
 
+    fn impl_from_form(&self) -> TokenStream2 {
+        if self.input_ident().as_deref() != Some("MultipartFormData") {
+            return quote! {};
+        }
+
+        let Some((field_name, _)) = self.single_field() else {
+            return quote! {};
+        };
+
+        let server_fn_path = self.server_fn_path();
+        let struct_name = self.struct_name();
+
+        quote! {
+            impl ::leptos::form::FromFormData for #struct_name {
```

Comment:
I'd suggest that this should be handled in a similar way to the `server_fn_path`: have a `from_form_data` field in the `ServerFnCall` struct that takes `Option<Path>`, which allows leptos to provide `::leptos::form::FromFormData`, but would allow other crates to provide their own path here. If `from_form_data` is `None`, omit this implementation.